app/mockServices/mockProjects.service.ts

```diff
@@ -0,0 +1,52 @@
+import * as angular from 'angular';
+
+interface IProjectsService {
+  canCreate(): angular.IPromise < any >;
+  get(projectName: string): angular.IPromise < any >;
+  update(projectName: string, data: any): angular.IPromise < any >;
+}
+
+export class ProjectsService implements IProjectsService {
+  public static $inject = ['$q'];
+
+  private $q : any;
+
+  constructor ($q: angular.IQService) {
+    this.$q = $q;
+  }
+
+  public canCreate() {
+    let deferred = this.$q.defer();
+
+    setTimeout(() => {
```

Comment:
We use `$timeout` to trigger a digest loop that rerenders the views.

We use `setTimeout` specifically when we know the view does not need to be reevaluated (e.g., updating duration values or updating c3 charts using the c3.js API)